### DISCUSSION TITLE
MMM Incremental Volume by time period

### DISCUSSION THREAD

**Discussion by allisonwiener** [pymc-labs/pymc-marketing#330] [Q&A]
I'm looking to get incremental volume (AKA contribution from each media variable) for all variables for two time periods (latest 52wks vs previous 52 wks). Any tips for how to do this with the MMM models? Thanks

**Comment by cetagostini** [comment]
Hey @allisonwienerclorox! I think you have two ways to attack this problem, some more sophisticated than others.

Since you are talking about 52 weeks, I am going to assume that you want to compare the contributions of year X with respect to the following year. And you are **implicitly assuming** that these contributions during each year could vary. Right?

In my opinion, the purest and most sophisticated way to do this is through variable coefficients in your model, if you want to have an idea, you can take a look at the post created by @juanitorduz (https://juanitorduz.github.io/bikes_pymc/) where he talks about this. Additionally, you have other resources like:
- [Fast and Flexible Bayesian Inference in Time-varying Parameter Regression Models](https://www.tandfonline.com/doi/full/10.1080/07350015.2021.1990772)
- [A Bayesian Approach To Estimation Of Time-Varying Regression Coefficients](https://www.nber.org/system/files/chapters/c9941/c9941.pdf)

This first option would require some understanding of how Gaussians process works and would require more computational power.

Another option, definitely less sophisticated and reliable, is to create two models, one training with only the first year and the other with the following year.  By doing this you could easily appreciate how the coefficients vary based on the different training periods. Looking at the estimates for each year separately is a reasonable descriptive strategy, I believe. 

Doing this you could even estimate your regression model over different "windows" of your data. For example, if you have yearly data from 2000 to 2020, you might first estimate your model using only data from 2000 to 2005, then from 2001 to 2006, then from 2002 to 2007, and so on. By doing this, you can see how your coefficients change over time as different years of data are included in the estimation. This method can be particularly useful for detecting changes in relationships over time.

Another approach, **if you are already controlling by seasons**, uses dummy variables to account for the time effect in your regression model. Essentially, it allows you to control for any year-specific effects that could be influencing your dependent variable. In this model, you would include dummy variables for each year in your dataset. After estimating this model, you can compare the coefficients on your variable of interest across different years.

Definitely, this topic is quite interesting and it is something that many of us are exploring now, there is no perfect answer. All methods have pros and cons, but I think the ideal is to have an idea rather than have a perfect idea!

Let me know if helps :)
